R-package/demo/multiclass_custom_objective.R

```diff
@@ -20,18 +20,20 @@ valids <- list(train = dtrain, test = dtest)
 # Method 1 of training with built-in multiclass objective
 # Note: need to turn off boost from average to match custom objective
 # (https://github.com/microsoft/LightGBM/issues/1846)
+params <- list(
+    min_data = 1L
+    , learning_rate = 1.0
+    , num_class = 3L
+    , boost_from_average = FALSE
+    , metric = "multi_logloss"
+)
 model_builtin <- lgb.train(
-    list()
+    params = params
```

Comment:
Is it possible to pass keyword arguments before positional ones in R?
```suggestion
    params
```

Reply:
yes, R doesn't care about that difference. R's argument parsing will first match all values passed as keyword args, then pass remaining positional arguments to remaining unmatched parameters.

```r
f <- function(x1, x2,  x3) {
    print(paste0("x1: ", x1, " | x2: ", x2, " | x3: ", x3))
}

f("a", "b", "c")
# [1] "x1: a | x2: b | x3: c"

f(x2 = "a", "b", "c")
# [1] "x1: b | x2: a | x3: c"

f("b", x3 = "a", "c")
# [1] "x1: b | x2: c | x3: a"
```

I'm fine with the suggestion, but just wanted to show R isn't opinionated about that the way Python is.

Reply:
Oh, interesting! Thanks for the explanation and sorry for the false alarm.